firebase-firestore/src/main/java/com/google/firebase/firestore/remote/AndroidConnectivityMonitor.java

```diff
@@ -96,53 +99,60 @@ public void run() {
     }
   }
 
+  private void configureBackgroundStateListener() {
+    BackgroundDetector.getInstance().addListener(this);
+  }
+
+  @Override
+  public void onBackgroundStateChanged(boolean background) {
+    if (!background && isConnected()) {
+      raiseCallbacks(/* connected= */ true);
```

Comment:
Is invoking the callbacks enough to reset the backoff? (I presume it is, just checking)

Reply:
Yeah, they already call `inhibitBackoff`.